src/ucp/core/ucp_context.c

```diff
@@ -581,13 +608,18 @@ static ucs_status_t ucp_fill_config(ucp_context_h context,
     context->config.request.size    = params->request_size;
     context->config.request.init    = params->request_init;
     context->config.request.cleanup = params->request_cleanup;
+#if ENABLE_MT
+    context->config.thread_mode     = params->thread_mode;
+#else
+    context->config.thread_mode     = UCS_THREAD_MODE_SINGLE;
```

Comment:
we need to return ERR_UNSUPPORTED in this case - otherwise use may see failures (because of no thread-safety) and don't know why